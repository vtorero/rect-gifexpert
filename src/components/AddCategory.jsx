import { useState } from "react";
export const AddCategory = ({ onAgregarCategoria }) => {
  const [inputValue, setinputValue] = useState();

  const onImputChage = ({ target }) => {
    setinputValue(target.value);
  };

  const onSubmit = (event) => {
    event.preventDefault();
    if (inputValue.trim().length <= 1) return;
    onAgregarCategoria(inputValue.trim());
    // setCategories(cat =>[inputValue,...cat]);
    setinputValue("");
  };

  return (
    <form onSubmit={onSubmit}>
      <input
        type="text"
        placeholder="Buscar Gifts"
        value={inputValue}
        onChange={onImputChage}
      />
    </form>
  );
};

