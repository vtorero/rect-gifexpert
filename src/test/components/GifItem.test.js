import { render,screen } from "@testing-library/react";
import { GifItem } from "../../components/GifItem";


describe('Prueba en el componente <GifItem/>', () => {
    const tittle ='Saitama';
    const url ='https://google.com/'

    test('Debe de hacer match con el snapshot', () => {




    const {container} = render(<GifItem title={tittle} url={url}/>);
    expect(container).toMatchSnapshot();

    })


    test('Debe de salir error sy no recibe el titulo y url', () => {

        render(<GifItem title={tittle} url={url}/>);
       expect(screen.getByRole('img').src).toBe(url+'ss');


     })
 })