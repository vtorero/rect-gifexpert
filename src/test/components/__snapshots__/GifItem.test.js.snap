// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en el componente <GifItem/> Debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitama"
      src="https://google.com/"
    />
    <p>
      Saitama
    </p>
  </div>
</div>
`;
